### ISSUE TITLE
🐛 BUG: not available @astrojs/parser in Safari due to regular expressions

### ISSUE THREAD

**Issue by togami2864** [snowpackjs/astro#1429]
### What package manager are you using?

npm v7.21.0

### What operating system are you using?

Mac 

### Describe the Bug

https://github.com/snowpackjs/astro/blob/2f902aa1c0aae5836007d293a7ab0b09b0c6b8d9/packages/astro-parser/src/parse/state/codespan.ts#L6
[↑The lookbehind syntax is not supported by safari. ](https://caniuse.com/js-regexp-lookbehind)

When I try to run @astrojs/parser in safari, I get an error.
<img width="883" alt="shot" src="https://user-images.githubusercontent.com/62130798/134767513-2b59c94a-43ac-428f-830d-1284da69ae08.png">

I believe that correcting this will resolve the problem.

In addition, the current parser seems to regard single backslashes( \ )  **as mere characters** , **not escape characters**.
The proof is that if you put in two backslashes, it considers there to be two backslashes instead of one.
<img width="1057" alt="scrennshot" src="https://user-images.githubusercontent.com/62130798/134767727-122cefcb-5d83-48b1-a209-2de17081af31.png">

https://astro-parser-ast-viewer.netlify.app/#N4Igzg9grgTgxgUxALhAHQ2gBmgdpAWwQAI4IATBPLEAXyA
Normally, "\\" would be interpreted as a single backslash "\".

In other words, if there is even one backquote, I don't think there can be any pattern that doesn't match the regular expression described above.

### Steps to Reproduce

1. git clone https://github.com/togami2864/astro-parser-ast-viewer.git
2. npm i
3. npm run dev in safari
4. Error!

### Link to Minimal Reproducible Example (Optional)

https://astro-parser-ast-viewer.netlify.app/#N4Igzg9grgTgxgUxALhCAvkA

**Comment by matthewp** [comment]
Thanks! I'm not sure we ever intended `@astrojs/parser` to be runnable in a browser. As we are preparing a new compiler there will be a new parser API coming out of that. Going to mark this as blocked on that release.

**Comment by FredKSchott** [comment]
Thanks for the report! This parser has been replaced by a new parser that you can find here: https://github.com/snowpackjs/astro-compiler-next
